src/sudoku-firestarter.js

```diff
@@ -44,55 +44,111 @@ exports.allPossibleValues = function(sudokuNumbers, row, column) {
  */
-exports.getExistingNumbersInQuadrant = function (sudokuNumbers, rowIndex) {
-    var quadrant = [];
+exports.getExistingNumbersInQuadrant = function (sudokuNumbers, row, column) {
+
+    var quadrant = [],
+        row = this.getFirstColumnOrRowOfQuadrant(row),
+        max = this.getMaxRowOrColumn(row);
 
-    for(var r = rowIndex; r <= 2; r++) {
-        quadrant.push(this.readQuadrantRow(sudokuNumbers, r));
+    for (var r = row; r <= max; r++) {
+        quadrant.push(this.readQuadrant(sudokuNumbers, r, column, max));
     }
 
-    return [].concat.apply([],quadrant);
+    // Flatten result of nested arrays to a single array
+    return [].concat.apply([], quadrant);
 };
 
+/**
+ * Returns a max index.
+ * TODO: Make this more flexible not fixed on to a sudoku with 3x3 quadrants.
+ *
+ * @param {Number} index which will make sure that no more than 3 items will be looped through.
+ * @param {Number} length max length of possible indexes.
+ *
+ * @returns {Number} will always return 2.
+ */
+exports.getMaxRowOrColumn = function(index, length) {
+    var i = index + 2;
+
+    for(var j = length; j > 0; j--) {
+
+        if(i % 3 === 0) {
+            return i;
+        } else {
+            i = i--;
+        }
+    }
+
+    return i;
+};
 
 /**
  * Function that will only traverse through a row within a quadrant.
  *
  * @param { Array } sudokuNumbers representing the whole sudoku game. Each array represents a row.
- * @param { Number } rowIndex the row within the sudokuNumbers that needs to be reviewed.
+ * @param { Number } row the row within the sudokuNumbers that needs to be reviewed.
+ * @param { Number } column the column within the sudokuNumbers that needs to be reviewed.
  *
  * @returns {Array} Array of numbers within one single row of a quadrant representing in a quadrant.
  */
-exports.readQuadrantRow = function (sudokuNumbers, rowIndex) {
-    var max = 2,
+exports.readQuadrant = function (sudokuNumbers, row, column) {
+
+    var max,
         quadrant = [];
 
-    for (var j = 0; j <= max; j++) {
-        if (this.isValidNumber(sudokuNumbers[rowIndex], sudokuNumbers[rowIndex][j])) {
-            quadrant.push(sudokuNumbers[rowIndex][j]);
+    // Start at the first cell of the quadrant.
+    column = this.getFirstColumnOrRowOfQuadrant(column);
+    max = this.getMaxRowOrColumn(column);
+
+    for (var c = column; c <= max; c++) {
+
+        if (this.isValidNumber(sudokuNumbers[row], sudokuNumbers[row][c])) {
+            quadrant.push(sudokuNumbers[row][c]);
         }
     }
 
     return quadrant;
 };
 
 /**
+ * Retrieves the index of the column or row where the quadrant starts.
+ *
+ * @param {Number} index of column of row.
+ * @returns {*}
+ */
+exports.getFirstColumnOrRowOfQuadrant = function (index) {
+    var diff = index;
+
+    if (index % 3 === 1) {
+
+        diff = index - 1;
+
+    } else if (index % 3 === 2) {
+
+        diff = index - 2;
```

Comment:
the code above looks pretty much the same like this one, you reckon you can simplify this a bit more?